### ISSUE TITLE
'autohide_bookmarks_and_main_toolbars.css' breaks search-bar on New Tab page

### ISSUE THREAD

**Issue by zZombieBatman** [MrOtherGuy/firefox-csshacks#441]
On the new tabs page there is a search-bar that when typed into it will normally send your input to the URLbar, but when the navBar is hidden due to the autohide, it won't send that input over to it at all.

**Comment by MrOtherGuy** [comment]
This should be fixed in 7c38557

**Comment by PavloVinnyk** [comment]
> This should be fixed in [7c38557](https://github.com/MrOtherGuy/firefox-csshacks/commit/7c38557addd67d43e27b391f5a4689a9c6504b9d)

With just this code in userChrome.css the urlbar renders outside interface.
![image](https://github.com/user-attachments/assets/f7bc0a79-3936-4241-9a90-30d6241ca398)

**Comment by MrOtherGuy** [comment]
Sheesh. Seems like it's only broken in beta, but works correctly in release and nightly. What fun...

**Comment by MrOtherGuy** [comment]
I dunno, kinda seems that beta is currently in a funny state where urlbar bounds are not computed correctly (at least not correctly for our purposes). You can get somewhat better behavior if you add this snippet to the style `#urlbar[popover]{ --urlbar-height: 32px !important; }` but it urlbar will still be in the wrong position initially when new window is created, but it should move into place once focused and typed something into it.

Most entertainingly, toolbar heights can get wrong, *but only sometimes* when window is created. So if a window is affected, then web-content will partially overlap with tabs toolbar. That's like caused by some bit of javascript sometimes running a bit too late or early when window is being created.

I guess chances are that something regarding this will get changed in a later beta build, I'm sort of thinking that whatever causes this style to fail here would also cause some other urlbar related issues unless fixed.

**Comment by PavloVinnyk** [comment]
Thank you for help, it's ok solution for me. Will wait future browser updates.

**Comment by zZombieBatman** [comment]
I applied the userChrome.css fixes to both the Nightly and Dev Build(beta), but the URL bar didn't behave as expected on either. This is due to my titlebar+navar having reduced sized which I always accommodate for by changing '--uc-navbar-height' to -32px.  However, the URL bar doesn't align properly. It places the #urlbar[popover] in what would be its default location

dev build (after launch):
![dev before](https://github.com/user-attachments/assets/5f811227-68ba-43dc-9dca-fff5afb0a32e)


dev build (post-initial focus):
![dev after](https://github.com/user-attachments/assets/8f603371-8b9b-48ed-9ff9-ab74646934c3)


nightly build:
![nightly](https://github.com/user-attachments/assets/7c92bda8-1e2d-41a9-8608-9ee2214cd290)

**Comment by PavloVinnyk** [comment]
In standard beta 133.0b2 it behaves like you described in initial message - window starts with shifted urlbar and moves to the right position with any input. And I checked that `#urlbar[popover]{ --urlbar-height: 32px !important; }` not needed to shift to right possition - not sure how i missed it initially.

**Comment by MrOtherGuy** [comment]
@zZombieBatman yeah it should be pretty  safe to remove those `--uc-urlbar-top` uses in nightly - I added them for beta only really and I'm now really confused why sometimes things seem to work fine without them. But, as previously mentioned there appears to be some seemingly non-deterministic behavior on beta so I can't make too much conclusion in one way or another.

@PavloVinnyk The panels actually should have been preventing toolbars from hiding this whole time, but apparently I have just had them do that when *hovered* - I'll fix that soon.
